barman/backup.py

```diff
@@ -618,6 +618,16 @@ def _validate_incremental_backup_configs(self):
                 "to perform an incremental backup using the Postgres backup method"
             )
 
+        parent_backup_id = kwargs.get("parent_backup_id")
+        parent_backup_info = self.get_backup(parent_backup_id)
+
+        if parent_backup_info and parent_backup_info.summarize_wal != "on":
```

Comment:
The `parent_backup_info` is expected to always be different of `None` at this point because `parse_backup_id` in the CLI already takes care of that.

But keeping the `if parent_backup_if` part of the `if` statement shouldn't hurt either.